### ISSUE TITLE
Option for passing PID file as FD instead of filename string?

### ISSUE THREAD

**Issue by akamensky** [sevlyar/go-daemon#54]
Right now PID is passed as filename string, and this limits what can be done with PID.

I think it would be good idea to give an option to pass PID or filename, if filename, then system takes care of making a PID file and locking it, otherwise it would be up to the developer to do that.

**Comment by sevlyar** [comment]
You are right. Good idea. But it is not simply to implement safe. And implementation of v2 is not going well as i want.
